lib/utils/under-dash.js

```diff
@@ -95,6 +97,46 @@ const _ = {
     if (a > b) return 1;
     return 0;
   },
+
+  isUndefined(val) {
+    return toString.call(val) === '[object Undefined]';
+  },
+
+  isArray(val) {
+    return toString.call(val) === '[object Array]';
+  },
+
+  isObject(val) {
+    return toString.call(val) === '[object Object]';
+  },
+
+  deepMerge() {
```

Comment:
what is the difference between `Object.assign()`?

Reply:
The `Object.assign()` method copies all enumerable own properties from one or more source objects to a target object. It returns the target object.The property value of the source object is a reference of an object, so it only points to that reference.

This `deepmerge()` will be deeply copied to the target object, similar to the concept of deep cloning.